src/pytest_servers/s3.py

```diff
@@ -23,7 +23,11 @@ def __enter__(self):
             if r.ok:
                 raise RuntimeError("moto server already up")
         self.proc = subprocess.Popen(
-            shlex.split("moto_server s3 -p %s" % self.port)
+            shlex.split(
+                f"moto_server s3 -p {self.port}",
+            ),
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
```

Comment:
Looks like we are seriously polluting your machine ports, which will turn ugly. Take a look at how we do it in dvc - we spawn one container with moto (and azurite in the future, for example) to simulate real s3 and we use a random single port without polluting.

Reply:
https://github.com/iterative/dvc-s3/tree/main/dvc_s3/tests

Reply:
If we at least don't spam lots of servers - this will become minor 🙂

Reply:
Added an optional flag (`--moto-server`) to set a custom port. In #13 I introduce a docker fixture which can be used to spawn docker containers. I will introduce some proper port management after that